ui/components/app/assets/nfts/nfts-items/nfts-items.js

```diff
@@ -116,39 +118,40 @@ export default function NftsItems({
     dispatch,
   ]);
 
-  const history = useHistory();
+  const getAssetImageUrlAndUpdate = async (image, nft) => {
+    const nftImage = await getAssetImageURL(image, ipfsGateway);
+    const updatedNFt = {
+      ...nft,
+      ipfsImageUpdated: nftImage,
+    };
+    return updatedNFt;
+  };
 
-  const renderCollectionImage = (collectionImage, collectionName) => {
-    if (collectionImage?.startsWith('ipfs') && !ipfsGateway) {
-      return (
-        <div className="nfts-items__collection-image-alt">
-          {collectionName?.[0]?.toUpperCase() ?? null}
-        </div>
-      );
-    }
-    if (!openSeaEnabled && !collectionImage?.startsWith('ipfs')) {
-      return (
-        <div className="nfts-items__collection-image-alt">
-          {collectionName?.[0]?.toUpperCase() ?? null}
-        </div>
-      );
-    }
+  useEffect(() => {
+    const promisesArr = [];
+    const modifyItems = async () => {
+      for (const key of collectionsKeys) {
+        const { nfts } = collections[key];
+        for (const singleNft of nfts) {
+          const { image, imageOriginal } = singleNft;
 
-    if (collectionImage) {
-      return (
-        <img
-          alt={collectionName}
-          src={getAssetImageURL(collectionImage, ipfsGateway)}
-          className="nfts-items__collection-image"
-        />
-      );
-    }
-    return (
-      <div className="nfts-items__collection-image-alt">
-        {collectionName?.[0]?.toUpperCase() ?? null}
-      </div>
-    );
-  };
+          const isImageHosted =
+            image?.startsWith('https:') || image?.startsWith('http:');
```

Comment:
Same thing here about the [URL ](https://developer.mozilla.org/en-US/docs/Web/API/URL/protocol) constructor (still no big deal and doesn't need to block)